### PR TITLE
feat(HTTP Request Tool Node): Use DynamicStructureTool with models supporting it (no-changelog)

### PR DESCRIPTION
**Pull request by burivuhster**
## Summary

This PR is an attempt to make passing parameters to HTTP tool more reliable. More specifically – to utilize model's tool calling feature when it is possible (e.g. OpenAI, Claude). To achieve this, we need to provide HTTP tool as a DynamicStructuredTool to the agent when it makes sense. To save backwards compatibility we need to provide DynamicAgent, as it was done before.

We introduced a wrapper class (N8nTool), extending DynamicStructuredTool, having an additional `asDynamicTool()` method, doing the wrapping.

Agents use HTTP Tool node as DynamicStructuredTool or DynamicTool depending on their abilities (see `getConnectedTools`)

To make sure we don't break existing workflows, we incremented the minor node version.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/AI-249/tools-agent-with-openai-chat-model-sometimes-refuses-to-pass-proper

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/ToolHttpRequest.node.ts [hunk 1]
```diff
@@ -394,9 +394,14 @@ export class ToolHttpRequest implements INodeType {
...
-		const description = prepareToolDescription(toolDescription, toolParameters);
+		const schema = makeToolInputSchema(toolParameters);
 
-		const tool = new DynamicTool({ name, description, func });
+		const tool = new N8nTool(this, {
+			name,
+			description: toolDescription,
+			func,
+			schema,
+		});
```
Comment (michael-radency) [inline packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/ToolHttpRequest.node.ts:405]: are we sure that after update behavior would be same? if not I would suggest to put those changes under new minor version

Comment (burivuhster) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/ToolHttpRequest.node.ts:405]: Do you mean the node version?

Comment (michael-radency) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/ToolHttpRequest.node.ts:405]: @burivuhster 
yes, exactly

**Review by michael-radency — APPROVED** [review]
LGTM :+1:

**Comment by janober** [comment]
Got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)

**Pull request merged**